src/tokens/Assets.tsx

```diff
@@ -0,0 +1,442 @@
+const HEADER_OPACITY_ANIMATION_START_OFFSET = 44
+// distance in points over which the screen header opacity animation is applied
+const HEADER_OPACITY_ANIMATION_DISTANCE = 20
+
+function AssetsScreen({ navigation, route }: Props) {
+  const { t } = useTranslation()
+
+  const activeTab = route.params?.activeTab ?? AssetTabType.Tokens
+
+  const supportedNetworkIds = getSupportedNetworkIdsForTokenBalances()
+  const tokens = useTokensWithTokenBalance()
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const totalTokenBalanceLocal = useTotalTokenBalance() ?? new BigNumber(0)
+  const tokensAreStale = useTokenPricesAreStale(supportedNetworkIds)
+
+  const insets = useSafeAreaInsets()
+
+  // TODO: Update this to filter out unsupported networks once positions support non-Celo chains
+  const positions = useSelector(positionsSelector)
+  const showPositions = getFeatureGate(StatsigFeatureGates.SHOW_POSITIONS)
+  const displayPositions = showPositions && positions.length > 0
+
+  const dappShortcutsEnabled = getFeatureGate(StatsigFeatureGates.SHOW_CLAIM_SHORTCUTS)
+  const positionsWithClaimableRewards = useSelector(positionsWithClaimableRewardsSelector)
+  const showClaimRewards =
+    dappShortcutsEnabled &&
+    positionsWithClaimableRewards.length > 0 &&
+    activeTab !== AssetTabType.Collectibles
+
+  // TODO: Update these to filter out unsupported networks once positions support non-Celo chains
+  const totalPositionsBalanceUsd = useSelector(totalPositionsBalanceUsdSelector)
+  const totalPositionsBalanceLocal = useDollarsToLocalAmount(totalPositionsBalanceUsd)
+  const totalBalanceLocal = totalTokenBalanceLocal?.plus(totalPositionsBalanceLocal ?? 0)
+
+  const [nonStickyHeaderHeight, setNonStickyHeaderHeight] = useState(0)
+  const [listHeaderHeight, setListHeaderHeight] = useState(0)
+  const [listFooterHeight, setListFooterHeight] = useState(0)
+
+  const scrollPosition = useSharedValue(0)
+  const footerPosition = useSharedValue(0)
+  const handleScroll = useAnimatedScrollHandler<{ prevScrollY: number }>(
+    {
+      onScroll: (event, ctx) => {
+        const scrollY = event.contentOffset.y
+        scrollPosition.value = scrollY
+
+        function clamp(value: number, min: number, max: number) {
+          return Math.min(Math.max(value, min), max)
+        }
+
+        // Omit overscroll in the calculation
+        const clampedScrollY = clamp(
+          scrollY,
+          0,
+          event.contentSize.height - event.layoutMeasurement.height
+        )
+
+        // This does the same as React Native's Animated.diffClamp
+        const diff = clampedScrollY - ctx.prevScrollY
+        footerPosition.value = clamp(footerPosition.value + diff, 0, listFooterHeight)
+        ctx.prevScrollY = clampedScrollY
+      },
+      onBeginDrag: (event, ctx) => {
+        ctx.prevScrollY = event.contentOffset.y
+      },
+    },
+    [listFooterHeight]
+  )
+
+  const animatedListHeaderStyles = useAnimatedStyle(() => {
```

Comment:
This just disables the shadow if `nonStickyHeaderHeight` is 0 and sets a shadow if it's visible? What triggers that change to happen?